src/Services/QueuedJobService.php

```diff
@@ -1086,9 +1286,21 @@ protected function handleBrokenJobException(QueuedJobDescriptor $jobDescriptor,
             $e->getMessage(),
             ['exception' => $e]
         );
+        $this->processBrokenJobOnException($jobDescriptor, $job, $e);
+        $jobDescriptor->write();
+    }
+
+    protected function processBrokenJobOnException(
+        QueuedJobDescriptor $jobDescriptor,
+        QueuedJob $job,
+        Throwable $e
+    ): void {
+        // Capture exception in the messages of the broken job for better debug options
+        $message = sprintf('%s : %s', $e::class, $e->getMessage());
+        $job->addMessage($message);
```

Comment:
This is DY code cleanup with the only functional change being the added message log.

Reply:
This can be split out of this PR if needed however it does help with testing.

Reply:
I think I would prefer this to be done separately, if at all. There's additional considerations to make (e.g. could this expose some private details which might be in an exception message?) that I'd prefer to deal with separately since this is already a very large PR with lots to consider.